tools/setup/emoji/build_emoji

```diff
@@ -384,6 +384,22 @@ def generate_map_files(cache_path: str, emoji_catalog: Dict[str, List[str]]) ->
             )
         )
 
+    # This is the more official API for mobile to fetch data about emoji.
+    # emoji_codes.json has a lot of goo, and we're creating this new file as a cleaner data format to move towards.
+    # We could add the rest of the data into this API-described data, and then the web client could switch to that
+    # which would allow us to drop the existing file. But we'll probably instead do #18121 which will make this file
+    # obsolete. So this is a temporary solution. CZO discussion:
+    # https://chat.zulip.org/#narrow/stream/378-api-design/topic/currently.20supported.20emoji/near/1394598
+    EMOJI_API_FILE_PATH = os.path.join(cache_path, "emoji_api.json")
+    with open(EMOJI_API_FILE_PATH, "wb") as emoji_api_file:
+        emoji_api_file.write(
+            orjson.dumps(
+                {
+                    "name_to_code": name_to_codepoint,
+                }
+            )
+        )
```

Comment:
Is it easy to move this commit to a separate PR? I'll post a few other comments on some API changelog details with this, but I think it can be merged independently/before the rest of these changes, and it'll simplify things to do that.